client/common/client.c

```diff
@@ -164,6 +169,13 @@ void freerdp_client_context_free(rdpContext* context)
 	if (!context)
 		return;
 
+#ifdef WITH_SSO_MIB
```

Comment:
did miss this, if you allocate in `freerdp_client_start` then you need to deallocate in `freerdp_client_stop` and not in `freerdp_client_free`

Reply:
ok, did run this and implemented the use of the common callback here https://github.com/akallabeth/FreeRDP/tree/sso-mib  (feel free to cherry pick the commit to your branch)

what I did find is this though:
* `mib_public_client_app_new` uses stuff from `rdpSettings` and therefore should only be called after `PreConnect` callback and not here.

So, I'd suggest you do allocate the `mibClientWrapper` in `ContextNew` and free it in `ContextFree` but only initialize it on first use of the callback.

Reply:
Just so I get it right:

If I move the allocation/freeing of `client_context->mibClientWrapper` into `freerdp_client_context_new/free`, I don't want to call `mib_public_client_app_new` there as the settings will not have been initialized. Instead, you want this initialized in which callback?

Inside `sso_mib_get_access_token` itself so a client (like `xfreerdp`) would:
- set `sso_mib_get_access_token` as their primary `GetAccessToken` callback and
- call `freerdp_set_common_access_token` with `client_cli_get_access_token`?

Or do I get the meanings of the new callback mixed up here?

Reply:
@rupran so, in short yes (sorry, did misread it first :grin: )